src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts

```diff
@@ -0,0 +1,262 @@
+            e.stopPropagation();
+            e.preventDefault();
+            this.showFilter($el[0]);
+        });
+
+        $el.appendTo(args.node);
+
+        //@ts-ignore
+        this.columnButtonMapping[column.id] = $el[0];
+    }
+
+    private showFilter(filterButton: HTMLElement) {
+        let $menuButton;
+        const target = withNullAsUndefined(filterButton);
+        if (target) {
+            $menuButton = jQuery(target);
+            this.columnDef = $menuButton.data("column");
+        }
+
+        const offset = jQuery(filterButton).offset();
+        if (offset) {
+            // If there is not enough vertical space under the filter button, we will move up the menu.
+            // const menuTop = offset.top + this.menu.offsetHeight <= window.innerHeight ? offset.top : window.innerHeight - this.menu.offsetHeight;
+            // Make sure the menu is on the screen horizontally.
+            // const menuLeft = offset.left + filterButton.offsetWidth + this.menu.offsetWidth <= window.innerWidth ? offset.left + filterButton.offsetWidth : window.innerWidth - this.menu.offsetWidth;
+        }
+
+        var $popup = jQuery(
+            '<div id="popup-menu">' +
+                '<button id="sort-ascending" type="button" icon="slick-header-menuicon.ascending" class="sort-btn">Sort Ascending</button>' +
+                '<button id="sort-descending" type="button" icon="slick-header-menuicon.descending" class="sort-btn">Sort Descending</button>' +
+                "</div>",
+        );
+
+        if (offset) {
+            $popup.css({
+                top: offset.top + $menuButton?.outerHeight()!, // Position below the anchor
+                left: offset.left, // Align left edges
+            });
+            // If there is not enough vertical space under the filter button, we will move up the menu.
+            // const menuTop = offset.top + this.menu.offsetHeight <= window.innerHeight ? offset.top : window.innerHeight - this.menu.offsetHeight;
+            // Make sure the menu is on the screen horizontally.
+            // const menuLeft = offset.left + filterButton.offsetWidth + this.menu.offsetWidth <= window.innerWidth ? offset.left + filterButton.offsetWidth : window.innerWidth - this.menu.offsetWidth;
+        }
+
+        $popup.appendTo(document.body);
+        openPopup($popup);
+        jQuery(document).on("click", (e: JQuery.ClickEvent) => {
+            // const $popup = jQuery('#popup-menu');
+            const $target = jQuery(e.target);
+
+            // If the clicked target is not the button or the menu, close the menu
+            if (
+                !$target.closest("#anchor-btn").length &&
+                !$target.closest("#popup-menu").length
+            ) {
+                $popup.fadeOut();
+            }
+        });
+
+        // Close the pop-up when the close-popup button is clicked
+        jQuery(document).on("click", "#close-popup", function () {
```

Comment:
Is this `close-popup` button defined/used anywhere?

Reply:
Yes, I just added it as an additional option to the menu - good catch.